app/order-by.pipe.ts

```diff
@@ -0,0 +1,18 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Course} from './course'
+
+@Pipe({
+    name: 'orderByDate'
+})
+
+export class OrderByDatePipe implements PipeTransform {
+    constructor() {}
+
+    transform(courses: Course[]): Course[]  {
+        courses.sort((a: any, b: any) => {
```

Comment:
native `sort` function is mutative — it modified the array it's applied to. This is bad for your application as it makes this pipe to be impure. Try using something like `lodash/sortBy` to achieve the result. It will allow you to make this pipe pure.